server.hpp

```diff
@@ -53,22 +56,59 @@ using std::chrono::milliseconds;
 using std::chrono::seconds;
 using std::chrono::system_clock;
 
+#ifdef __GNUC__
+#include <range/v3/all.hpp>
+#else
+namespace ranges = std::ranges;
+#endif
+
 constexpr auto TIMEOUT { 30s };
 
 class Room;
 
 void start_session(asio::io_context&, Room&, asio::error_code&, std::string_view, std::string_view);
 
+struct ContestRequest {
+    enum class Result {
+        ACCEPTED,
+        REJECTED,
+        WAITING,
+    };
+
+    Participant_ptr sender;
+    Participant_ptr receiver;
+    Role role;
+    Result result;
+    std::chrono::system_clock::time_point time;
+
+    ContestRequest(Participant_ptr sender, Participant_ptr receiver, Role role)
+        : sender { sender }
+        , receiver { receiver }
+        , role { role }
+        , result { Result::WAITING }
```

Comment:
You can assign a default value when defining the field.

Reply:
I'm going to remove the `result` field. It is unused.